ovos_utils/parse.py

```diff
@@ -1,6 +1,106 @@
 from difflib import SequenceMatcher
 import re
 from inflection import singularize as _singularize_en
+from enum import IntEnum, auto
+from ovos_utils.log import LOG
+
+try:
+    import rapidfuzz
+except ImportError:
+    rapidfuzz = None
+
+
+class MatchStrategy(IntEnum):
+    SIMPLE_RATIO = auto()
+    RATIO = auto()
+    PARTIAL_RATIO = auto()
+    TOKEN_SORT_RATIO = auto()
+    TOKEN_SET_RATIO = auto()
+    PARTIAL_TOKEN_RATIO = auto()
+    PARTIAL_TOKEN_SORT_RATIO = auto()
+    PARTIAL_TOKEN_SET_RATIO = auto()
+    QUICK_LEV_RATIO = auto()
+
+
+def _validate_matching_strategy(strategy):
+    if rapidfuzz is None and strategy != MatchStrategy.SIMPLE_RATIO:
+        LOG.error("rapidfuzz is not installed, "
+                  "falling back to SequenceMatcher for all match strategies")
+        LOG.warning("pip install rapidfuzz")
+        return MatchStrategy.SIMPLE_RATIO
+    return strategy
+
+
+def fuzzy_match(x, against, strategy=MatchStrategy.SIMPLE_RATIO):
```

Comment:
I'm getting some very high scores for unrelated strings:

```python3
>>> parse.fuzzy_match("four score and seven years ago", "i am a banana")
0.18604651162790697
>>> parse.fuzzy_match("four score and seven years ago", "queen elizabeth ii")
0.3333333333333333
>>> parse.fuzzy_match("who is the current president of mexico", "intentional blocking")
0.27586206896551724
>>> parse.fuzzy_match("who is the current president of mexico", "personal computer")
0.18181818181818182
```

Is it just me, or should most of those be lower, and shouldn't the low ones be near zero?

Reply:
i would argue anything bellow 0.5 should be discarded.

how do you feel making 0.5 the new zero? so those would actually be negative

Reply:
I dunno. I like making these scores lower/count for less, but it might break the use of ovos_utils as a drop-in replacement, because there might be skills (potentially lots of skills) with hardcoded values.

Reply:
not meant as a drop in replacement either way ;)

Reply:
re the scores, those essentially reflect the number of edits to turn one string into the other, so if we want that to mean how close the strings are to each other then maybe we do need to think a bit better about it. out of scope for this PR but interested to pick your brains about it